src/main/java/com/okta/tools/OktaAwsCliAssumeRole.java

```diff
@@ -94,8 +97,23 @@ private static Path getSessionFilePath() {
 
     String run(Instant startInstant) throws Exception {
         Optional<Session> session = getCurrentSession();
-        if (session.isPresent() && sessionIsActive(startInstant, session.get()))
-            return session.get().profileName;
+        Instant expiry = startInstant.plus(864000 - 30, ChronoUnit.SECONDS);
+        String profileStore = System.getProperty("user.home") + "/.aws/.okta-profile-expiry";
+        Ini ini = new Ini(new File(profileStore));
```

Comment:
This is the line that introduces the crash I saw. You can't assume the session file exists.

You probably shouldn't even know that it's a file from this method. The whole interaction with your new named profile session expiry persistence (the INI file) should be moved to a class of its own.